libzkbob-rs/src/address.rs

```diff
@@ -154,36 +138,51 @@ fn parse_address_raw<P: PoolParams>(
     }
 }
 
-// generates shielded address in format "pool_prefix:base58(d ++ p_d ++ checksum)"
+// generates shielded address in format "base58(d ++ p_d ++ checksum)"
+// pool prefix doesn't append here
+// both address types (pool-specific\generic) can generated here
 pub fn format_address<P: PoolParams>(
     d: BoundedNum<P::Fr, { constants::DIVERSIFIER_SIZE_BITS }>,
     p_d: Num<P::Fr>,
-    pool: Option<Pool>, // set pool to None to generate universal address for all pools
+    pool_id: Option<u32>, // set pool_id to None to generate universal address for all pools
 ) -> String {
     let mut buf: [u8; ADDR_LEN] = [0; ADDR_LEN];
 
     d.serialize(&mut &mut buf[0..10]).unwrap();
     p_d.serialize(&mut &mut buf[10..42]).unwrap();
 
     // there are two ways for checksum calculation
-    let (checksum_hash, address_prefix) = match pool {
+    let checksum_hash = match pool_id {
         // pool-specific format
-        Some(pool) => {
+        Some(pool_id) => {
             const POOL_ID_BYTES: usize = POOL_ID_BITS >> 3;
             let mut hash_src: [u8; POOL_ID_BYTES + 32] = [0; POOL_ID_BYTES + 32];
-            //let pool_id = pool.pool_id_num::<P::Fr>().to_num().to_uint().0.to_big_endian();
-            //let pool_id_be: [u8; POOL_ID_BYTES] = pool_id[32 - POOL_ID_BYTES..32].try_into().unwrap();
-            pool.pool_id_bytes_be::<P::Fr>().serialize(& mut &mut hash_src[0..POOL_ID_BYTES]).unwrap();
+            pool_id_to_bytes_be::<P>(pool_id).serialize(& mut &mut hash_src[0..POOL_ID_BYTES]).unwrap();
             hash_src[POOL_ID_BYTES..POOL_ID_BYTES + 32].clone_from_slice(&keccak256(&buf[0..42]));
-            (keccak256(&hash_src), pool.address_prefix().to_owned())
+            keccak256(&hash_src)
         },
         // generic format (for all pools, when pool_id isn't specified)
-        None => (keccak256(&buf[0..42]), GENERIC_ADDRESS_PREFIX.to_string()),
+        None => keccak256(&buf[0..42]),
     };
     buf[42..ADDR_LEN].clone_from_slice(&checksum_hash[0..4]);
 
     let address_part = bs58::encode(buf).into_string();
 
-    format!("{}:{}", address_prefix, address_part)
+    format!("{}", address_part)
+
+}
+
+fn pool_id_to_bytes_be<P: PoolParams>(pool_id: u32) -> [u8; POOL_ID_BYTES] {
+    /*/ preparing pool id for checksum validation
+    const POOL_ID_BYTES: usize = POOL_ID_BITS >> 3;
+    let pool_id_num = Num::<P::Fr>::from_uint(NumRepr(Uint::from_u64(pool_id as u64))).unwrap();        
+    let pool_id_bytes = pool_id_num.to_uint().0.to_big_endian();
+    let pool_id_be: [u8; POOL_ID_BYTES] = pool_id_bytes[32 - POOL_ID_BYTES..32].try_into().unwrap();*/
+
+    // preparing pool id for checksum validation
+    const POOL_ID_BYTES: usize = POOL_ID_BITS >> 3;
```

Comment:
I'm not 100 % sure but it looks like a complicated way to do:
```
pool_id.to_be_bytes()[4 - POOL_ID_BYTES..].try_into().unwrap()
```

Reply:
Oh, thank yo for that point! It's the best way to do that and it works. Fixed in https://github.com/zkBob/libzkbob-rs/pull/77/commits/80e7c87e21d8501d0355f7146e9b4cb5f9f8fcb7